src/main/webapp/scripts/serverclient.js

```diff
@@ -40,23 +40,39 @@ class ServerClient {
      * @private {URLSearchParams}
      */
     this.urlParams_ = urlParams;
+
+    this.startSessionCache_();
   }
 
   /** 
    * This method starts to cache.
    * @private
    */
   startSessionCache_() {
-    throw new Error('Unimplemented');
+    this.sessionCache_.start();
   }
 
   /**
    * This method changes the controller of the current session to the 
-   * Attendee pased in.
-   * @param {string} name 
+   * Attendee passed in.
+   * @param {string} newControllerName 
    */
-  passController(name) {
-    throw new Error('Unimplemented');
+  changeController(newControllerName) {
+    const /** string */ sessionID = 
+        encodeURI(this.urlParams_.get('session-id'));
+    const /** Request */ request = new Request('/change-controller', {
```

Comment:
Could you create private constants in this file of the names of the endpoints? For example:

const CHANGE_CONTROLLER_ENDPOINT_ = '/change-controller';
..
..
.. new Request(CHANGE_CONTROLLER_ENDPOINT, .. )

Do these for other endpoints that you use

Reply:
i am having trouble doing the same for the get-session endpoint, since the scope of the function is local it doesn't have access to any of the class variables

Reply:
Create it here and pass it to the SessionCache in the constructor or something. Does that work?

Reply:
ohh wait i think i misunderstood initially, you meant to define these variables in the serverclient file not within the class itself

Reply:
right. define it in this file and pass it to the SessionCache via the constructor. Basically, dont export it outside of this file